modules/libcom/src/osi/os/posix/osdThread.c

```diff
@@ -326,13 +330,22 @@ int          status;
     a_p->usePolicy = arg.ok;
 }
 #endif
-
+
+static int childAfterFork;
+
+static void childHook(void)
+{
+    epicsAtomicSetIntT(&childAfterFork, 1);
+}
 
 static void once(void)
 {
     epicsThreadOSD *pthreadInfo;
     int status;
 
+    status = pthread_atfork(NULL, NULL, &childHook);
+    checkStatusOnce(status, "pthread_atfork");
```

Comment:
According to [this documentation](https://docs.rtems.org/branches/master/posix-users/process_creation_and_execution.html#pthread-atfork) RTEMS will return `ENOSYS` from this `pthread_atfork()` call, which will trigger a warning message.

Reply:
Ok, I expected I'd have to `#ifdef` this for one reason or another.

Reply:
You could avoid using an `#ifdef` just for RTEMS by accepting a status value of `ENOSYS` from any OS since that tells us whatever OS it is probably doesn't support forking, or at least can't do `pthread_atfork()`.

Reply:
It's a conditional one way or the other.  I think `#ifdef __rtems__` is clearer.